caikit/core/data_model/dataobject.py

```diff
@@ -199,7 +214,7 @@ def decorator(cls: Type) -> Type[DataBase]:
 
         # Declare the merged class that binds DataBase to the wrapped class with
         # this generated proto class
-        if isinstance(proto_class, type):
+        if not isinstance(proto_class, EnumTypeWrapper):
```

Comment:
Took me a bit to catch what was going on here, but I think this is right

Reply:
Otherwise you get:
<img width="873" alt="image" src="https://github.com/caikit/caikit/assets/6641871/b79b3d8e-b41c-4515-961d-6719afe4974c">

Which I think the `if isinstance(proto_class, type)` forces it to be a type, and you would need to do a cast down or something to pass it to that function.